lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -1241,10 +1243,28 @@ private String safeLockStats() {
     @Override
     public void close() {
         if (isShutDown.compareAndSet(false, true)) {
-            lockReapRunner.close();
-            threadInfoSnapshotManager.close();
-            blockingThreads.forEach(Thread::interrupt);
-            callOnClose.run();
+            List<Pair<String, Runnable>> closures = List.of(
+                    Pair.create("lockReapRunner", lockReapRunner::close),
+                    Pair.create("threadInfoSnapshotManager", threadInfoSnapshotManager::close),
+                    Pair.create("blockingThreads", () -> blockingThreads.forEach(Thread::interrupt)),
+                    Pair.create("callOnClose", callOnClose));
+
+            List<Throwable> encounteredThrowables = new ArrayList<>();
```

Comment:
not actionable: It might be ok to leave this for now in the interest of readability, though the allocation on the happy path does trip my suspicion detectors.